ostd/libs/linux-bzimage/setup/src/x86/amd64_efi/decoder.rs

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! This module is used to decompress payload.
+
+extern crate alloc;
+
+pub use alloc::vec::Vec;
+
+use core2::io::Read;
+use libflate::{deflate, gzip, zlib};
+
+pub fn decompress_payload(payload: &[u8], magic: &[u8]) -> Vec<u8> {
```

Comment:
You should not let the user of this function provide the magic. In this function you can read it from the payload slice. Also, add a simple documentation to the function describing what it does and what it supports.